install.sh

```diff
@@ -1,41 +1,156 @@
+    \"install_method\":\"$INSTALL_METHOD\",
+    \"started_at\":\"$STARTED_AT\",
+    \"ended_at\":\"$ENDED_AT\",
+    \"outcome\":\"$OUTCOME\",
+    \"step_failed\":\"$STEP_FAILED\",
+    \"dont_track_ip\":true
+  }
+}"
+    [ -n "${SHUTTLE_DEBUG:-}" ] && echo -e "Sending telemetry data:\n$telemetry_data"
+    curl -sL -H 'Content-Type: application/json' -d "$telemetry_data" https://console.shuttle.dev/ingest/i/v0/e > /dev/null
+  fi
+}
+
+_exit_success() {
+  OUTCOME="success"
+  _send_telemetry
+  echo
+  [[ "$SUPPORTS_COLOR" = "1" ]] && echo -en "\e[32m" # green
+  echo "Thanks for installing Shuttle CLI! 🚀"
+  [[ "$SUPPORTS_COLOR" = "1" ]] && echo -en "\e[0m"
+  exit 0
+}
+
+_exit_neutral() {
+  echo
+  echo "If you have any problems, please open an issue on GitHub or visit our Discord!"
+  exit 0
+}
+
+_exit_failure() {
+  STEP_FAILED="$1"
+  OUTCOME="fail"
+  echo
+  [[ "$SUPPORTS_COLOR" = "1" ]] && echo -en "\e[31m" # red
+  echo "Shuttle installation script failed with reason: $STEP_FAILED"
+  [[ "$SUPPORTS_COLOR" = "1" ]] && echo -en "\e[0m"
+  echo "If you have any problems, please open an issue on GitHub or visit our Discord!"
+  _send_telemetry
+  exit 1
+}
 
 if ! command -v curl &>/dev/null; then
-  echo "curl not installed. Please install curl."
-  exit
+  echo "curl not installed. Please install curl or use a different install method."
+  _exit_failure "curl-not-found"
 elif ! command -v sed &>/dev/null; then
-  echo "sed not installed. Please install sed."
-  exit
+  echo "sed not installed. Please install sed or use a different install method."
+  _exit_failure "sed-not-found"
 fi
 
 REPO_URL="https://github.com/shuttle-hq/shuttle"
-LATEST_RELEASE=$(curl -L -s -H 'Accept: application/json' "$REPO_URL/releases/latest")
+LATEST_RELEASE=$(curl -fsL -H 'Accept: application/json' "$REPO_URL/releases/latest")
+[ $? -ne 0 ] && _exit_failure "check-latest-release"
 # shellcheck disable=SC2001
 LATEST_VERSION=$(echo "$LATEST_RELEASE" | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+[ $? -ne 0 ] && _exit_failure "parse-latest-version"
+
+if command -v cargo-shuttle &>/dev/null; then
+  NEW_INSTALL="false"
+  if [[ "$(cargo-shuttle -V)" = *"${LATEST_VERSION#v}" ]]; then
+    [[ "$SUPPORTS_COLOR" = "1" ]] && echo -en "\e[32m" # green
+    echo "Shuttle CLI is already at the latest version!"
+    [[ "$SUPPORTS_COLOR" = "1" ]] && echo -en "\e[0m"
+    exit 0 # skip telemetry and instantly exit
+  else
+    echo "Updating Shuttle CLI to $LATEST_VERSION"
+  fi
+fi
 
 _install_linux() {
   echo "Detected Linux!"
   echo "Checking distro..."
   if (uname -a | grep -qi "Microsoft"); then
     OS="ubuntuwsl"
   elif ! command -v lsb_release &>/dev/null; then
-    echo "lsb_release could not be found. Falling back to /etc/os-release"
+    [ -n "${SHUTTLE_DEBUG:-}" ] && echo "lsb_release could not be found. Falling back to /etc/os-release"
     OS="$(grep -Po '(?<=^ID=).*$' /etc/os-release | tr '[:upper:]' '[:lower:]')" 2>/dev/null
```

Comment:
Perl regexps are a bit unorthodox, and some versions of grep (e.g. alpine/busybox) do not support them.
Here is something that should be more compatible:
```suggestion
    OS="$(grep -Eo '^ID=.*$' /etc/os-release | cut -d '=' -f 2 | tr '[:upper:]' '[:lower:]')" 2>/dev/null
```